### PR TITLE
Add HTML email template string to config (gsi 19)

### PR DESCRIPTION
**Pull request by TheByronHimes**
Adds the notifier port and implementation, with basic templating support for email construction from a template.

### CODE REVIEW

File: ns/core/notifier.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    def __init__(self, *, config: Config):
+        """Initialize the Notifier with configured host, port, and so on"""
+        self._config = config
+
+    async def send_notification(self, *, notification: event_schemas.Notification):
+        """Sends notifications based on the channel info provided (e.g. email addresses)"""
+        raise NotImplementedError
+
+    def _construct_email(self, *, notification: event_schemas.Notification):
```
Comment (KerstenBreuer) [inline ns/core/notifier.py:38]: A bit nicer to have that documented:
```suggestion
    def _construct_email(self, *, notification: event_schemas.Notification) -> EmailMessage:
```

File: ns/core/notifier.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        message["To"] = notification.recipient_email
+        message["Cc"] = notification.email_cc
+        message["Bcc"] = notification.email_bcc
+        message["Subject"] = notification.subject
+
+        # concatenate recipient name and plaintext body with generic greeting/closing
+        plaintext_email = (
+            f"Dear {notification.recipient_name},\n\n"
+            + f"{notification.plaintext_body}\n\nWarm regards,\n\nThe GHGA Team"
+        )
```
Comment (KerstenBreuer) [inline ns/core/notifier.py:50]: This is an alternative to the HTML version?

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Great. Looks very clean. Just some minor things.

File: ns/core/notifier.py (hunk 1 above)
Comment (TheByronHimes) [reply ns/core/notifier.py]: Added it

File: ns/core/notifier.py (hunk 2 above)
Comment (TheByronHimes) [reply ns/core/notifier.py]: It is done

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**